### PR TITLE
Expand CVAR landing page to include discovery information

### PR DESCRIPTION
**Pull request by lmbr-pip**
Signed-off-by: Pip Potter <61438964+lmbr-pip@users.noreply.github.com>

## Change summary

Added content I often forget about when working with CVARs including how to discover CVARs in the project via the console window.

There is a little repetition here of other content, but where possible I linked to that content.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/appendix/cvars/_index.md [hunk 1]
```diff
@@ -1,27 +1,56 @@
 ---
-title: CVARs
+title: Console variables (CVARs)
```
Comment (AMZN-Gene) [inline content/docs/user-guide/appendix/cvars/_index.md:2]: Yes!

File: content/docs/user-guide/appendix/cvars/_index.md [hunk 2]
```diff
@@ -1,27 +1,56 @@
...
 
 You can invoke CVARs in the Editor via the [console](/docs/user-guide/editor/console/).
 
 You can also pass CVARs on the command line when launching O3DE binaries via `--{cvar name} {value}` or `--{cvar name}={value}`. For example, `ServerLauncher.exe --sv_somebool=true`.
 
+Additionally, you can add CVARs to [configuration files](/docs/user-guide/editor/console/#configuring-console-variables-in-configuration-files).
+
+### Discovering CVARs
+
+The easier way to find the CVARs settable in the current project is via the [console variables window](/docs/user-guide/editor/console/#viewing-the-console-window). Bringing up the console window will list all the CVARs in the current project.
```
Comment (AMZN-Gene) [inline content/docs/user-guide/appendix/cvars/_index.md:19]: When I read "the easier way" it sounded like it implied there was another (less easy?) method for finding cvars above. This is the only way to find cvar described, so maybe i'd say... 
"The way to find CVARs..." or 
"The easiest way to find CVARS..." or 
"Besides searching the code base for CVAR, the easier way to find CVARs..."

**Review by AMZN-Gene — APPROVED** [review]

File: content/docs/user-guide/appendix/cvars/_index.md (hunk 2 above)
Comment (lmbr-pip) [reply content/docs/user-guide/appendix/cvars/_index.md:19]: You can discover cvars via the dump cvar commands and through the auto complete functionality in the console, so this is the easier way, but I can definitely rewrite.

**Review by allisaurus — APPROVED** [review]
Great addition, super helpful

File: content/docs/user-guide/appendix/cvars/_index.md [hunk 3]
```diff
@@ -1,27 +1,56 @@
...
 
 You can also pass CVARs on the command line when launching O3DE binaries via `--{cvar name} {value}` or `--{cvar name}={value}`. For example, `ServerLauncher.exe --sv_somebool=true`.
 
+Additionally, you can add CVARs to [configuration files](/docs/user-guide/editor/console/#configuring-console-variables-in-configuration-files).
+
+### Discovering CVARs
+
+The fastest way to find all the CVARs settable in the current project is via the [console variables window](/docs/user-guide/editor/console/#viewing-the-console-window). Bringing up the console window will list all the CVARs in the current project as well as allow you to set their values.
+
+You can also begin to type the start of CVAR name in the console and then use `tab` to show autocomplete and look at all the potential matches. For example, after typing `ed_` you can use the tab key to cycle through all the CVARs that impact the editor.
```
Comment (willihay) [inline content/docs/user-guide/appendix/cvars/_index.md:21]:
```suggestion
You can also begin to type the start of a CVAR name in the console and then use the **Tab** key to show autocomplete and look at all the potential matches. For example, after typing `ed_` you can use **Tab** to cycle through all the CVARs that impact the editor.
```

File: content/docs/user-guide/appendix/cvars/_index.md [hunk 4]
```diff
@@ -1,27 +1,56 @@
...
+
+![Automatic CVAR expansion in the console](/images/user-guide/appendix/cvars/console_autoexpand.png)
+
+### CVAR console functions
+
+There are built-in console commands that make working with CVARs easier:
+
+| Name             | Description                                                                                                                                                   |
+|------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| DumpCommandsVars | (Legacy) Will dump all the CVARs and their values to a file called `consolecommandsandvars.txt`.<br/>Prefer to use the console window to see CVARs and values |
```
Comment (willihay) [inline content/docs/user-guide/appendix/cvars/_index.md:31]:
```suggestion
| DumpCommandsVars | (Legacy) Will dump all the CVARs and their values to a file called `consolecommandsandvars.txt`.<br>Prefer to use the console window to see CVARs and values. | 
```

File: content/docs/user-guide/appendix/cvars/_index.md [hunk 5]
```diff
@@ -1,27 +1,56 @@
...
+![Automatic CVAR expansion in the console](/images/user-guide/appendix/cvars/console_autoexpand.png)
+
+### CVAR console functions
+
+There are built-in console commands that make working with CVARs easier:
+
+| Name             | Description                                                                                                                                                   |
+|------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| DumpCommandsVars | (Legacy) Will dump all the CVARs and their values to a file called `consolecommandsandvars.txt`.<br/>Prefer to use the console window to see CVARs and values | 
+| resetcvars       | Resets all CVARs to their initial values                                                                                                                      |
```
Comment (willihay) [inline content/docs/user-guide/appendix/cvars/_index.md:32]:
```suggestion
| resetcvars       | Resets all CVARs to their initial values.                                                                                                                     |
```

File: content/docs/user-guide/appendix/cvars/_index.md [hunk 6]
```diff
@@ -1,27 +1,56 @@
...
-| sv_    | For server only CVARs.                                                       |
+O3DE uses the following prefixes to make discovering CVARs easier:
+
+| Prefix | Usage                                                                            |
+|--------|----------------------------------------------------------------------------------|
+| bg_    | **B**oth **G**ames, for common CVARs that can be used on both client and server. |
+| cl_    | For client only CVARs.                                                           |
+| ed_    | For editor only CVARs.                                                           |
+| net_   | For low-level networking CVARs.                                                  |
+| physx_ | For physx related CVARs.                                                         |
```
Comment (willihay) [inline content/docs/user-guide/appendix/cvars/_index.md:44]: NVIDIA requirement
```suggestion
| physx_ | For NVIDIA PhysX related CVARs.                                                  |
```

File: content/docs/user-guide/appendix/cvars/_index.md [hunk 7]
```diff
@@ -1,27 +1,56 @@
...
+| r_     | For rendering related CVARs.                                                     |
+| s_     | For sound related CVARs                                                          |
```
Comment (willihay) [inline content/docs/user-guide/appendix/cvars/_index.md:46]:
```suggestion
| s_     | For sound related CVARs.                                                         |
```

**Review by willihay — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/appendix/cvars/_index.md (hunk 4 above)
Comment (micronAMZN) [inline content/docs/user-guide/appendix/cvars/_index.md:31]: When I run this  command on my development build, the file is written to `./Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/consolevars.txt` even when the DCCSI Gem is not active in the project. Can users specify where to write this file?

**Review by micronAMZN — APPROVED** [review]
Approved pending willihay's feedback.

File: content/docs/user-guide/appendix/cvars/_index.md (hunk 4 above)
Comment (lmbr-pip) [reply content/docs/user-guide/appendix/cvars/_index.md]: I believe its based on your project path, will have to check.

Comment (lmbr-pip) [reply content/docs/user-guide/appendix/cvars/_index.md]: You can't set the location, its based on the current directory of the process. For builds, its the build folder ie build\windows_vs2019\bin\profile etc.

Comment (lmbr-pip) [reply content/docs/user-guide/appendix/cvars/_index.md]: The code doesn't take a path or directory, it just says write the file 'consolevars.txt' without any pathing ie

```
azfopen(&f0, "consolecommandsandvars.txt", "w");
```

Platform OS versions of fopen/fopen_s will be invoked and these set relative paths relative to the current working directory to the process invoked.

From a documentation POV, theres nothing you can currently set to control pathing. So I would suggest requesting this as a feature/reporting it as a bug, but theres nothing we can add to these docs right now.

**Review by willihay — APPROVED** [review]

**Pull request merged**